### DISCUSSION TITLE
Add <Client>.addCommand() or <Client>.setCommands()

### DISCUSSION THREAD

**Discussion by DanieliusDev** [discordjs/discord.js#7471] [Ideas]
add a function to Client that can make a Application or Guild command based on what you put.
examle:
```js
const client = new Client(/* options */);

client.addCommand({ // example of adding the command data that is used to register slash commands to discord
    name: 'ping',
    description: 'The ping command'
}, (interaction) => { // example of how you would execute the command
    interaction.reply({ content: 'Pong!' })
}, { // example of some setting for the command
    cooldown: 5,
    guildCommand: true
});
```
this could simplify the command handling process because this function would push the command to discord for you, also you can add a registerCommands() function  Client so the user can call this function to push the commands to discord when they have finished adding their commandts to the client.

I am currently making a Project that i have made this feature in, i can send the github for it when im done with it.

**Comment by monbrey** [comment]
No thank you. We already have functions available to register commands in `client.application.commands.create()` and `guild.commands.create()`.

Command handling is not a core component of discord.js and should instead be handled in frameworks like Sapphire.
